src/firmware/app/control/trajectory_planner.h

```diff
@@ -72,232 +71,369 @@ typedef enum TrajectoryPlannerGenerationStatus
+ * and profile of the trajectory.
+ *
+ * @return A status indicating whether or not generation was successful
  */
 TrajectoryPlannerGenerationStatus_t
-app_trajectory_planner_generateConstantArcLengthPositionTrajectory(
-    PositionTrajectory_t* trajectory);
+app_trajectory_planner_generateConstantParameterizationPositionTrajectory_2(
+    PositionTrajectory_t* position_trajectory,
+    FirmwareRobotPathParameters_t path_parameters);
 
 /**
- * Returns a constant interpolation period (time) trajectory based on an input trajectory
+ * Function that modifies an existing speed profile to be backwards continuous. This means
+ * that it is possible to reach every speed node within the limits of deceleration
  *
- * This function is intended to take a variable time trajectory (which is easier to
- * calculate) then interpolate a constant period trajectory. This is valuable for discrete
- * time motion controllers that operate only at constant delta-time intervals
+ * @pre All arrays are pre-allocated up to at least TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
  *
- * NOTE: This function does not modify the real trajectory. It will output a physically
- * invalid trajectory if the input trajectory is invalid
+ * @param speeds [in/out] The existing speed profile to be modified in-place to be
+ * backwards continuous
  *
- * @pre constant_period_trajectory is pre-allocated up the the limit specified by
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ * @param segment_lengths [in] The length of each segment in the trajectory
+ *
+ * @param max_allowable_acceleration [in] The maximum acceleration that can occur anywhere
+ * on the trajectory
+ *
+ * @param initial_speed [in] The initial speed at the beggining of the trajectory
  *
- * @param constant_period_trajectory [out] The array to be filled with the constant time
- * interpolation period equivalent of the input trajectory
- * @param variable_time_trajectory [in] The valid trajectory used as
- * reference for a constant interpolation period trajectory
- * @param interpolation_period [in] The constant change in time [s] that corresponds to
- * each trajectory segment.
+ * @param num_segments [in] The number of nodes(elements) in the trajectory
  *
+ * @return A status indicating whether or not generation was successful
  */
 TrajectoryPlannerGenerationStatus_t
-app_trajectory_planner_interpolateConstantPeriodPositionTrajectory(
-    PositionTrajectory_t* constant_period_trajectory,
-    PositionTrajectory_t* variable_time_trajectory, const float interpolation_period);
+app_trajectory_planner_modifySpeedsToBackwardsContinuous(
+    float speeds[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float segment_lengths[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float max_allowable_acceleration, float initial_speed, unsigned int num_segments);
 
 /**
- * Generates the X/Y points for each position on the constant arc length trajectory
+ * Function creates a forwards continuous speed profile based on the specified parameters.
```

Comment:
same as above, may be good to specify this is linear speed (not angular) just to be super clear

Reply:
Same as above, angular and linear use the same function in the new implementation

Reply:
I think it might be good to indicate explicitly for these that they can be used for linear or angular, I definitely had to look at the implementation here to figure out exactly how they're shared.

Reply:
Added a comment 👍